tasks/connectors/edgescan/lib/edgescan_vulnerability.rb

```diff
@@ -28,8 +28,8 @@ def location_specifiers
         end
 
         def matching_location_specifier
-          location_specifiers.find { |specifier| specifier["id"] == location_specifier_id } ||
-            location_specifiers.find { |specifier| location.include?(specifier["location"]) }
+          location_specifiers.find { |specifier| not_ip_range?(specifier) && specifier["id"] == location_specifier_id } ||
```

Comment:
Why are we checking to see if an IP address is in an IP address range rather than directly comparing IP addresses? 

Isn't there already a 1:1 mapping between hosts and locations?

Reply:
Yes there is a 1:1 mapping as you say for hosts and vulnerabilities. This is more for vulnerabilities that do not have a corresponding host.

There is not always a 1:1 mapping between vulnerabilities and location specifier's locations. A locations specifier could have a URL location "https://test.example.com", and the vulnerability's location could be "https://test.example.com/some/further/page.html". Though the location is different we would want to group those vulnerabilities with that location specifier together.

If a host is unavailable, then the matching location specifier's location will be used as the vulnerability's external ID, and I did not want an external ID with an IP address range in it.

For example:
Host is available: external_id => "ES1234 {host.location}"
Host is unavailable, but there is a matching location specifier => "ES1234 {location_specifier.location" unless the location is an IP address range
Host and location specifier are unavailable => "ES1234 {vulnerability.location}"